happiness-backend/api/dao/happiness_dao.py

```diff
@@ -64,7 +66,74 @@ def get_happiness_by_group_count(user_ids, page, n):
         Happiness.timestamp.desc()) \
         .paginate(page=page, per_page=n, error_out=False)
 
+def get_happiness_by_filter(user_id, page, per_page, start, end, low, high, query):
+    """
+    Filters according to the provided arguments. Checks to see what filters to apply. Will not apply the following
+    filters with the following arguments
+     low/high = 11
+     start/end = None
+     text = "".
+    """
+    if start is not None and low != 11 and query != "":
```

Comment:
why only check if start is not None and low != 11 without checking end and high? You are making the assumption these values are provided if low and start are provided. This assumption could cause an internal server error if unchecked.

Reply:
good point